backend/src/main/scala/com/softwaremill/adopttapir/starter/content/ContentMerger.scala

```diff
@@ -0,0 +1,45 @@
+package com.softwaremill.adopttapir.starter.content
+
+object ContentMerger {
+
+  def createTree(topDirName: String, pathNames: List[String], content: String): Directory = {
+    assert(pathNames.nonEmpty, "`pathNames` list cannot be empty!")
+
+    def innerCreateTree(pathNames: List[String], content: String): Node = {
+      pathNames match {
+        case Nil          => throw new IllegalStateException("pathNames list cannot be empty!")
+        case last :: Nil  => File(last, content)
+        case head :: tail => Directory(head, List(innerCreateTree(tail, content)))
+      }
+    }
```

Comment:
You are right, `innerCreateTree` could be defined outside `createTree` as a private method. There are no obvious gains here, I've used to use such inner methods when I need to do something before I could invoke a recursive call. Are there any drawbacks to using such an approach?

Reply:
It can be reviewed/examined in isolation (which is good cuz it is a method) yet it increases the complexity of `createTree` (caller). One can just put it before the caller so that it is visually _close_ to it (it answers the need of having definition just right before the code) and doesn't require much scrolling if needed...

Reply:
Moved `innerCreateTree` outside `createTree` as private `createTreeRec`